### PR TITLE
Webserver documentation Revision

### PR DESCRIPTION
**Pull request by ninjapanzer**
###### What am I?
Under review of the webserver-setup documentation I noticed it was a little unfocused and while the configurations were correct they did not contain line documentation.
There were a couple of discrepancies between windows and linux NGINX configurations. So I have combined them and kept the focus on linux with associated support for fast cgi and proxy_pass config.

I have also reordered the file to put the `built-in` server documentation first. No one is going to get to the NGINX or apache configuration until they have something to deploy in the meantime to improve framework adoption promoting the lowest weight runtime should be first.

Better yet this configuration is not platform dependent and should hopefully get users running without working about a server at all initially.

@niden I am looking for your feedback. This is still a WIP regarding the nginx configuration which I want to double check before this is merged.

### CODE REVIEW

**Comment by ninjapanzer** [comment]
This is WIP until I can confirm the NGINX configuration is without errors but the rest of the changes are ready to review.

**Comment by ninjapanzer** [comment]
I setup a test environment for NGINX to test the combined config and it has been updated to work.

File: en/webserver-setup.md [hunk 1]
```diff
@@ -11,82 +14,119 @@
...
-    }
+Then point your browser to http://localhost:8000/ to check if everything is working.
 
-    location ~ /\.ht {
-        deny all;
-    }
-}
-```
+<a name='nginx'></a>
+## NGINX
```
Comment (sergeyklay) [inline en/webserver-setup.md]: Could you please use common name like `Nginx` (not `NGINX`, `NginX`, etc) :)

File: en/webserver-setup.md [hunk 2]
```diff
@@ -11,82 +14,119 @@
...
+## PHP Built-In Webserver (For Developers)
+To speed up getting your Phalcon application running in development the easiest way is to use this built-in PHP server. Do not use this server in a production environment. The following configurations for [NGINX](#nginx) and [Apache](#apache) are what you need.
 
-    # This is the folder that index.php is in
-    root /var/www/phalcon/public;
-    index index.php index.html index.htm;
+<a name='php-built-in-phalcon-configuration'></a>
+### Phalcon configuration
+To enable dynamic URI rewrites, without Apache or NGINX, that Phalcon needs, you can use the following router file:
+<a href="https://github.com/phalcon/phalcon-devtools/blob/master/templates/.htrouter.php" target="_blank">.htrouter.php</a>
```
Comment (sergeyklay) [inline en/webserver-setup.md]:
```
[`.htrouter.php`](https://github.com/phalcon/phalcon-devtools/blob/master/templates/.htrouter.php)
```

[`.htrouter.php`](https://github.com/phalcon/phalcon-devtools/blob/master/templates/.htrouter.php)

**Review by sergeyklay — CHANGES_REQUESTED** [review]

File: en/webserver-setup.md (hunk 2 above)
Comment (ninjapanzer) [reply en/webserver-setup.md]: I would prefer to allow this link to target a new tab. If you still disagree I will update it.
Thanks

**Comment by ninjapanzer** [comment]
I have rebased this branch to remove merge conflicts due to navigation conversion from Markdown to html

**Review by sergeyklay — APPROVED** [review]
Cc: @niden

**Comment by niden** [comment]
Thank you!

**Pull request merged**